pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -597,8 +608,25 @@ func TestReconcile(t *testing.T) {
 	// Delete useraccount and ensure related resources are also removed
 	t.Run("delete useraccount removes subsequent resources", func(t *testing.T) {
 		// given
+
+		// when the member operator secret exists and has a che admin user configured then che user deletion is enabled
+		restore := test.SetEnvVarsAndRestore(t,
+			test.Env("WATCH_NAMESPACE", "toolchain-member"),
+			test.Env("MEMBER_OPERATOR_SECRET_NAME", "test-secret"),
+			// Che has two different routes for Che & Keycloak but CRW uses a single route for both.
+			// For tests we'll assume they're two different routes.
+			test.Env(configuration.MemberEnvPrefix+"_CHE_KEYCLOAK_ROUTE_NAME", "keycloak"),
+		)
+		defer restore()
+
+		defer gock.OffAll()
+		gockTokenSuccess()
+		gockFindUserSuccess("johnsmith")
+		gockDeleteUser(204)
```

Comment:
could we check that it was realy called? and the number of calls...?
for that you could add a function:
```
func SpyOnCalls(counter *int) gock.Matcher {
	matcher := gock.NewBasicMatcher()
	matcher.Add(func(_ *http.Request, _ *gock.Request) (bool, error) {
		*counter++
		return true, nil
	})
	return matcher
}
```
and use it:
```
func gockDeleteUser(code int, calls *int) {
	gock.New(testCheURL).
		Delete("api/user").
		SetMatcher(SpyOnCalls(calls)).
		MatchHeader("Authorization", "Bearer abc.123.xyz").
		Persist().
		Reply(code)
}
```
you could do the same also for the other gock mocks - since you use the `Persist()` method there and not `Times()`.

Reply:
That's a good idea! I've done that and added assertions for the counter. Thanks for the suggestion! 🥇